app/src/main/java/com/ojasvi/memify/MainActivity.kt

```diff
@@ -57,27 +61,36 @@ fun MainScreen(modifier: Modifier = Modifier) {
 }
 
 @Composable
-fun ImagePicker(modifier: Modifier = Modifier){
-    var imageUri by remember { mutableStateOf<Uri?>(null)}
-    
-    val imagePickerLauncher = rememberLauncherForActivityResult(contract = ActivityResultContracts.GetContent())
-    { uri: Uri? ->
+fun ImagePicker(modifier: Modifier = Modifier) {
+    var imageUri by remember { mutableStateOf<Uri?>(null) }
+
+    val pickMedia = rememberLauncherForActivityResult(ActivityResultContracts.PickVisualMedia()) { uri ->
+        // Callback is invoked after the user selects a media item or closes the
+        // photo picker.
         imageUri = uri
     }
-    
-    LaunchedEffect(Unit) {
-        imagePickerLauncher.launch("image/*")
-    }
-    
-    
+
+
     Column(
         horizontalAlignment = Alignment.CenterHorizontally,
         verticalArrangement = Arrangement.Center,
-        modifier = Modifier.fillMaxSize()
-    ){
+        modifier = Modifier.fillMaxSize().padding(20.dp)
+    ) {
+        Button(
+            onClick = { pickMedia.launch(PickVisualMediaRequest(ActivityResultContracts.PickVisualMedia.ImageOnly)) },
+            modifier = Modifier.fillMaxWidth().height(50.dp)
+        ) {
+            Text(text = "Open Gallery Picker")
+        }
+
         imageUri?.let { uri ->
-            Image(painter = rememberAsyncImagePainter(uri), contentDescription = null,
-                modifier = Modifier.size(200.dp).clip(CircleShape).border(2.dp, MaterialTheme.colorScheme.onBackground, CircleShape))
+            Image(
+                painter = rememberAsyncImagePainter(uri), contentDescription = null,
+                modifier = Modifier
+                    .size(200.dp)
```

Comment:
Try to use fillMaxWidth(0.5f) or something alone the lines. Fixed sizes are bad when you're running the app on different devices.

Reply:
Updated to use  fillMaxWidth(0.7f) as suggested